### PR TITLE
feat(button): Add ability to color icons separately from the text

### PR DESCRIPTION
**Pull request by williamernest**
Fixes: #2335
I added the `mdc-button-icon-color` mixin. The other mixin wasn't needed, since the icons use the text colors by default.

### CODE REVIEW

File: demos/button.html [hunk 1]
```diff
@@ -306,6 +306,20 @@ <h1 class="mdc-typography--display2">CSS Only</h1>
             </button>
           </div>
         </fieldset>
+
+        <fieldset>
+          <legend class="mdc-typography--title">Different Color Icons/Text</legend>
+          <div>
+            <button class="mdc-button mdc-button--unelevated demo-text-color" data-demo-no-js>
+              <i class="material-icons mdc-button__icon">favorite</i>
+              Big Corner Radius
```
Comment (kfranqueiro) [inline demos/button.html]: Update the labels on these buttons; these labels were copied from the previous two examples but don't make sense here.

File: demos/button.html [hunk 2]
```diff
@@ -306,6 +306,20 @@ <h1 class="mdc-typography--display2">CSS Only</h1>
             </button>
           </div>
         </fieldset>
+
+        <fieldset>
+          <legend class="mdc-typography--title">Different Color Icons/Text</legend>
+          <div>
+            <button class="mdc-button mdc-button--unelevated demo-text-color" data-demo-no-js>
```
Comment (bonniezhou) [inline demos/button.html]: Should we include a demo for JS-upgraded buttons too?

File: demos/button.scss [hunk 3]
```diff
@@ -44,3 +44,12 @@
 .mdc-button.thick-stroke-button {
   @include mdc-button-stroke-width(4px);
 }
+
+.demo-text-color {
+  @include mdc-button-icon-color(white);
+  @include mdc-button-ink-color(orange);
```
Comment (kfranqueiro) [inline demos/button.scss]: Should we also be applying mdc-states(orange) here? (I'm assuming that should still prioritize matching text over matching icon? Maybe we should ask design, if text/icon can be different colors, which color the "ink" and therefore the ripple is?)

File: demos/button.html [hunk 4]
```diff
@@ -234,6 +234,20 @@ <h1 class="mdc-typography--display2">Ripple Enabled</h1>
           </div>
         </fieldset>
 
+        <fieldset>
+          <legend class="mdc-typography--title">Different Color Icons/Text</legend>
```
Comment (kfranqueiro) [inline demos/button.html]: Also replace Text w/ Ink here

File: demos/button.html [hunk 5]
```diff
@@ -363,6 +377,20 @@ <h1 class="mdc-typography--display2">CSS Only</h1>
             </button>
           </div>
         </fieldset>
+
+        <fieldset>
+          <legend class="mdc-typography--title">Different Color Icons/Text</legend>
```
Comment (kfranqueiro) [inline demos/button.html]: and here

**Pull request merged**